include/server.h

```diff
@@ -52,7 +52,16 @@ namespace kiwix
        void stop();
 
        void setRoot(const std::string& root);
-       void setAddress(const std::string& addr) { if(!addr.empty()) addr.find('[') != std::string::npos ? m_addr.addr6 = addr : m_addr.addr = addr; }
+       void setAddress(const std::string& addr) {
+          if (!addr.empty()) {
+              if (addr.find('[') != std::string::npos) {
+                  std::string cleanAddr = addr.substr(1, addr.length() - 2);
+                  m_addr.addr6 = cleanAddr;
+              } else {
+                  m_addr.addr = addr;
+              }
+          }
+      }
```

Comment:
1. Move the definition to .cpp
2. Is `[` a mandatory part of an IPv6 address syntax? Isn't it better to test for the presence of a colon (`:`)? Of course we can remove optional square brackets, but we should also accept IPv6 addresses without brackets (like we did before).

Reply:
Exactly what I want do on next iteration. Thanks.